pkg/tfgen/generate_python.go

```diff
@@ -493,12 +493,6 @@ func (g *pythonGenerator) emitResourceType(mod *module, res *resourceType) (stri
 	w.Writefmtln(
 		`            warnings.warn("explicit use of __opts__ is deprecated, use 'opts' instead", DeprecationWarning)`)
 	w.Writefmtln("            opts = __opts__")
-	w.Writefmtln("        if not resource_name:")
-	w.Writefmtln("            raise TypeError('Missing resource name argument (for URN creation)')")
-	w.Writefmtln("        if not isinstance(resource_name, str):")
-	w.Writefmtln("            raise TypeError('Expected resource name to be a string')")
-	w.Writefmtln("        if opts and not isinstance(opts, pulumi.ResourceOptions):")
-	w.Writefmtln("            raise TypeError('Expected resource options to be a ResourceOptions instance')")
```

Comment:
not needed.  We already do this in `class Resource:`:

```
        if not name:
            raise TypeError(
                'Missing resource name argument (for URN creation)')
        if not isinstance(name, str):
            raise TypeError('Expected resource name to be a string')
```

Going to add the ` isinstance(opts, pulumi.ResourceOptions)` check to pulumi/pulumi.

Reply:
In order for this to be completely safe, we'll need to bump the minimum version of the Pulumi Python SDK to one that includes the `ResourceOptions` check.